### PR TITLE
[ET-1205] Fix ticket settings update breaking provider meta data.

### PR DESCRIPTION
**Pull request by rafsuntaskin**
### 🎫 Ticket

[ET-1205] 

### 🗒️ Description

* When the Ticket Settings update is clicked we trigger the `save_form_settings` method that makes use of `update_post_meta` for saving the provider data.
* For the new Ticket Commerce our provider class is namespaced which should be saved like this as meta: `TEC\\Tickets\\Commerce\\Module` but as it is running through `wp_unslash` it's getting saved as `TECTicketsCommerceModule`.

### 🎥 Artifacts 
🎥 screencast(s): https://d.pr/v/kOMioC

### ✔️ Checklist
- [ ] ~~I've included a changelog entry both in readme.txt.~~ ( Not required )
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1205]: https://theeventscalendar.atlassian.net/browse/ET-1205

### CODE REVIEW

File: src/Tickets/Commerce/Hooks.php [hunk 1]
```diff
@@ -416,4 +420,12 @@ public function add_display_post_states( $post_states, $post ) {
 		return $post_states;
 	}
 
+	public function skip_sanitization( $meta_value, $meta_key ) {
+
+		if ( $meta_value != wp_unslash( tribe( Module::class )->class_name ) ) {
+			return $meta_value;
+		}
+
+		return tribe( Module::class )->class_name;
```
Comment (bordoni) [inline src/Tickets/Commerce/Hooks.php:429]: `tribe( Module::class )->class_name` holds the same value as: `Module::class`

File: src/Tickets/Commerce/Hooks.php [hunk 2]
```diff
@@ -416,4 +420,12 @@ public function add_display_post_states( $post_states, $post ) {
 		return $post_states;
 	}
 
+	public function skip_sanitization( $meta_value, $meta_key ) {
+
+		if ( $meta_value != wp_unslash( tribe( Module::class )->class_name ) ) {
```
Comment (bordoni) [inline src/Tickets/Commerce/Hooks.php:425]: `tribe( Module::class )->class_name` holds the same value as: `Module::class`

File: src/Tickets/Commerce/Hooks.php [hunk 3]
```diff
@@ -416,4 +420,12 @@ public function add_display_post_states( $post_states, $post ) {
 		return $post_states;
 	}
 
+	public function skip_sanitization( $meta_value, $meta_key ) {
```
Comment (bordoni) [inline src/Tickets/Commerce/Hooks.php:423]: Doc block. functionality should not live in the Hooks class, so create a method on the `Settings` and use a filter here to point to it. See the method `filter_registration_cart_provider`

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tickets/Commerce/Hooks.php [hunk 4]
```diff
@@ -93,6 +93,10 @@ protected function add_filters() {
 
 		// Add a post display state for special Event Tickets pages.
 		add_filter( 'display_post_states', [ $this, 'add_display_post_states' ], 10, 2 );
+
+		// @todo make the following filters dynamic.
```
Comment (paulmskim) [inline src/Tickets/Commerce/Hooks.php:97]: could we add a person responsible for this + a jira ticket?

File: src/Tickets/Commerce/Settings.php [hunk 5]
```diff
@@ -397,4 +397,23 @@ public function apply_commerce_enabled_conditional( $settings ) {
...
+	 *
+	 * @since TBD
+	 *
+	 * @param string $meta_key   Metadata key.
+	 * @param mixed  $meta_value Metadata value.
+	 *
+	 * @return string
+	 */
+	public function skip_sanitization( $meta_value, $meta_key ) {
+
```
Comment (paulmskim) [inline src/Tickets/Commerce/Settings.php:411]: no need to pad functions with empty lines

File: src/Tickets/Commerce/Hooks.php [hunk 6]
```diff
@@ -416,4 +420,17 @@ public function add_display_post_states( $post_states, $post ) {
...
+	 * Handle saving of Ticket provider meta data.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $meta_key   Metadata key.
+	 * @param mixed  $meta_value Metadata value.
+	 *
+	 * @return string
+	 */
+	public function handle_provider_meta( $meta_value, $meta_key ) {
```
Comment (bordoni) [inline src/Tickets/Commerce/Hooks.php:433]: Since this is a filter, we should prefix it. `filter_modify_sanitization_provider_meta`

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Tickets/Commerce/Hooks.php (hunk 4 above)
Comment (bordoni) [reply src/Tickets/Commerce/Hooks.php:97]: Actually can we make it into a dynamic thing now? It shouldn't be hard.

**Review by bordoni — APPROVED** [review]

**Pull request merged**